File: src/com/designpatterns/creational/factory/ShapeFactory.java
package com.designpatterns.creational.factory;

public class ShapeFactory implements IShapeFactory {

    @Override
    public IShape getShape(String shapeType) {
        if (shapeType == null) {
            return null;
        }
        if (shapeType.equalsIgnoreCase(Circle.TAG)) {
            return new Circle(5.0f);
        } else if (shapeType.equalsIgnoreCase(Rectangle.TAG)) {
            return new Rectangle(4.0f, 5.0f);
        } else if (shapeType.equalsIgnoreCase(Square.TAG)) {
            return new Square(5.0f);
        } else {
            return null;
        }
    }
}

File: src/com/designpatterns/creational/factory/kotlin/ShapeFactory.kt
package com.designpatterns.creational.factory.kotlin

import java.lang.IllegalArgumentException

class ShapeFactory : IShapeFactory {

    override fun getShape(shapeType: String): IShape {

        return if (shapeType.equals(Circle.TAG, ignoreCase = true)) {
            Circle(5.0f)
        } else if (shapeType.equals(Rectangle.TAG, ignoreCase = true)) {
            Rectangle(4.0f, 5.0f)
        } else if (shapeType.equals(Square.TAG, ignoreCase = true)) {
            Square(5.0f)
        } else {
            throw IllegalArgumentException("Shape type not found")
        }
    }
}